src/components/TodoList.jsx

```diff
@@ -0,0 +1,32 @@
+import React, { useState } from "react";
+
+
+export default function TodoList({ tasks, setTasks, filter }) {
+  const visibleTasks = tasks.filter(t => {
+    if (filter === "All") return t
+    if (filter === "Deadline") return t.date !== ""
+    if (filter === "Completed") return t.done === true
+  })
```

Comment:
```suggestion
  const visibleTasks = filter === 'All' ? tasks : tasks.filter(t => {
    if (filter === "Deadline") return t.date
    if (filter === "Completed") return t.done
  })
```

1. Since we already have a full list, there's no need to run an additional cycle if `filter` is `All`. Instead, you can return the original `tasks` array, otherwise, run `filter` method
2. If `t.date` is `''`, Javascript will see it as a "falsy" value. This means it can be shortened to:
    ```js
    if (filter === "Deadline") return t.date
    ```

    > ... all values are truthy except `false`, `0`, `-0`, `0n`, `""`, `null`, `undefined`, `NaN`, and `document.all`
  
    Reference: https://developer.mozilla.org/en-US/docs/Glossary/Truthy
3. `t.done` is boolean itself, so you can return it directly, like this:

    ```js
    if (filter === "Completed") return t.done
    ```